straxen/url_config.py

```diff
@@ -0,0 +1,201 @@
+
+import json
+import strax
+import fsspec
+import straxen
+from numpy import isin
+import inspect
+from urllib.parse import urlparse, parse_qs
+from ast import literal_eval
+from functools import lru_cache
+
+from strax.config import OMITTED
+
+export, __all__ = strax.exporter()
+
+
+def parse_val(val):
+    try:
+        val = literal_eval(val)
+    except:
+        pass
+    return val
+
+
+@export
+class URLConfig(strax.Config):
+    """Dispatch on URL protocol.
+    unrecognized protocol returns identity
+    inspired by dasks Dispatch and fsspec fs protocols.
+    """
+    
+    _LOOKUP = {}
+    SCHEME_SEP = '://'
+    QUERY_SEP = '?'
+    PLUGIN_ATTR_PREFIX = 'plugin.'
+
+    def __init__(self, cache=False, **kwargs):
```

Comment:
perhaps add to a docstring that this is in bytes (and/or do a byes to MB conversion in the `if cache` block and rename the variable to `max_cache_mb`)